### PR TITLE
Add integration tests: Helm in the PolicyGenerator

### PR DESCRIPTION
**Pull request by dhaiducek**
Confirmed that `kustomize` internally uses `helm template`:
https://github.com/kubernetes-sigs/kustomize/blob/91ccf00ac8f787551eb4c5baad486b088ef3737b/api/types/helmchartargs.go#L149

Which doesn't call the Kubernetes API, so security here should be fine, but we have a `lookup` in the chart just to make sure:
```shell
$ helm template --help
Render chart templates locally and display the output.

Any values that would normally be looked up or retrieved in-cluster will be
faked locally. Additionally, none of the server-side testing of chart validity
(e.g. whether an API is supported) is done.
```

ref: https://issues.redhat.com/browse/ACM-9456

### CODE REVIEW

**Comment by dhaiducek** [comment]
I've upgraded the E2E cluster, so this PR should be good to merge once CI passes.

File: test/integration/policy_generator_helm_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+			g.Expect(err).ShouldNot(HaveOccurred(), "The subscription status should be parseable.")
+			g.Expect(found).Should(BeTrue(), "The subscription status should have a phase.")
+			g.Expect(appSubPhase).Should(Equal("Propagated"), "The subscription should propagate successfully.")
+		},
+			defaultTimeoutSeconds,
+			1,
+		).Should(Succeed())
+	})
+
+	It("Validates the propagated policies", func() {
```
Comment (yiraeChristineKim) [inline test/integration/policy_generator_helm_test.go:49]:
```suggestion
	It("Validates the propagated policies", func(ctx SpecContext) {
```

Optional

File: test/integration/policy_generator_helm_test.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+		// Perform some basic validation on the generated policy.
+		By("Checking that the root policy was created")
+		policyRsrc := clientHubDynamic.Resource(common.GvrPolicy)
+		var policy *unstructured.Unstructured
+		Eventually(
+			func() error {
+				var err error
+				policy, err = policyRsrc.Namespace(namespace).Get(
+					context.TODO(), policyName, metav1.GetOptions{},
```
Comment (yiraeChristineKim) [inline test/integration/policy_generator_helm_test.go:58]: Optional: ctx SpecContext
```suggestion
					ctx, policyName, metav1.GetOptions{},
```

**Review by yiraeChristineKim — DISMISSED** [review]

File: test/integration/policy_generator_helm_test.go (hunk 1 above)
Comment (dhaiducek) [reply test/integration/policy_generator_helm_test.go:49]: Oh wow. I didn't even realize that was a thing! Thanks!

**Review by yiraeChristineKim — APPROVED** [review]

**Pull request merged**